### PR TITLE
animal mimics

### PR DESCRIPTION
**Pull request by funkychicken493**
oh my god this is terrifying

### CODE REVIEW

File: denizen_scripts/HeroCraft/mobs/animal_mimic.dsc [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            - stop if:<context.damager.is_player.not.if_null[true]>
+            - determine passively cancelled
+            - define location <context.entity.location>
+            - remove <context.entity> if:<context.entity.exists>
+            - spawn "<entity[animal_mimic].with[custom_name=<&f><context.entity.name.to_titlecase> Mimic].with_flag[mimic_drops:<context.drops>]>" <[location]> target:<context.damager.if_null[<[location].find_entities[player].within[50].first>]> persistent
+
+animal_mimic:
+    type: entity
+    debug: false
+    entity_type: skeleton
```
Comment (Ph4i1ur3) [inline denizen_scripts/HeroCraft/mobs/animal_mimic.dsc]: Let's change this to something a bit more scary, like a creeper!

Comment (funkychicken493) [reply denizen_scripts/HeroCraft/mobs/animal_mimic.dsc]: Ok!

Comment (funkychicken493) [reply denizen_scripts/HeroCraft/mobs/animal_mimic.dsc]: I don't think a creeper is very scary, so I made it a husk.

**Comment by funkychicken493** [comment]
converted to draft because i need to add some more entities that can be mimics and more chance multipliers

File: denizen_scripts/HeroCraft/mobs/animal_mimic.dsc [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    events:
+        after cow|sheep|pig|chicken|rabbit|frog|llama|goat spawns because NATURAL:
+            - stop if:<context.location.town.exists>
+            - define chance 0.5
+            - define chance:*:2 if:<context.location.world.has_storm>
+            - define chance:*:2 if:<context.location.world.time.is_more_than_or_equal_to[13000]>
+            - define chance:*:2 if:<context.entity.entity_type.is_in[llama|goat]>
+            - if <util.random_chance[<[chance]>]>:
+                - flag <context.entity> animal_mimic expire:5m
+        on entity_flagged:animal_mimic dies:
```
Comment (Ph4i1ur3) [inline denizen_scripts/HeroCraft/mobs/animal_mimic.dsc:13]: I feel like it'd be better served as a damage or click event

File: denizen_scripts/HeroCraft/mobs/animal_mimic.dsc [hunk 3]
```diff
@@ -16,6 +16,11 @@ animal_mimic_events:
             - define location <context.entity.location>
             - remove <context.entity> if:<context.entity.exists>
             - spawn "<entity[animal_mimic].with[custom_name=<&f><context.entity.name.to_titlecase> Mimic].with_flag[mimic_drops:<context.drops>]>" <[location]> target:<context.damager.if_null[<[location].find_entities[player].within[50].first>]> persistent
+        on player clicks entity_flagged:animal_mimic:
```
Comment (Ph4i1ur3) [inline denizen_scripts/HeroCraft/mobs/animal_mimic.dsc]: right clicks, Funky
player right clicks entity

Comment (funkychicken493) [reply denizen_scripts/HeroCraft/mobs/animal_mimic.dsc]: WHAT

**Comment by funkychicken493** [comment]
@Ph4i1ur3 oh my lordie lord

**Pull request merged**